cdatgui/cdat/plotter.py

```diff
@@ -156,24 +157,33 @@ def can_plot(self):
 
     @property
     def canvas(self):
-        # print "MANAGER CANVAS:", self.source.canvas
         return self.source.canvas
 
     def gm(self):
-        return self._gm
+        return vcs.getgraphicsmethod(self._type, self._gm)
 
     def set_gm(self, gm):
-        # check gm vs vars
-        self._gm = gm
-        if self.can_plot():
+        plot = True
+        if isinstance(gm, tuple):
+            plot = gm[1]
+            gm = gm[0]
+
+        self._gm = gm.name
+        self._type = vcs.graphicsmethodtype(gm)
+        if plot and self.can_plot():
             self.plot()
+        self.source.gm_label.setText(self._gm)
 
     graphics_method = property(gm, set_gm)
 
     def get_vars(self):
         return self._vars
 
     def set_vars(self, v):
+        plot = True
```

Comment:
I really don't like this solution. Can you point me at the part of the code that is using this? I want to try and rethink how we're handling this.